sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -201,6 +205,15 @@ private[sql] class HDFSBackedStateStoreProvider extends StateStoreProvider with
     override def toString(): String = {
       s"HDFSStateStore[id=(op=${id.operatorId},part=${id.partitionId}),dir=$baseDir]"
     }
+
+    override def valuesIterator(key: UnsafeRow, colFamilyName: String): Iterator[UnsafeRow] = {
```

Comment:
May I ask is there any consideration of not throwing error for `valuesIterator` in `HDFSBackedReadStateStore` and throws error here in `HDFSBackedStateStore`?

Reply:
Hmm... I am not sure why I did this. We should throw an error here. I have modified the logic to throw an error instead.